shtdlib.sh

```diff
@@ -420,7 +429,7 @@ function compare_versions {
     assert [ ${#items[@]} -gt 0 ]
     #shellcheck disable=SC2119
     lowest_ver=$(printf "%s\\n" "${items[@]}" | version_sort | head -n1)
-    lowest_ver_line=$(printf "%s\\n" "${items[@]}" | grep --line-regexp "${lowest_ver}" --line-number --max-count=1 | awk -F: '{print $1}')
+    lowest_ver_line=$(printf "%s\\n" "${items[@]}" | grep -e "^${lowest_ver}$" -c | awk -F: '{print $1}')
```

Comment:
I needed `--max-count=1` because a list can have the same item twice.. so we should limit it with `grep` or another utility if compatibility is an issue

Reply:
Replaced the whole thing with a better approach.